src/Timer/Timers.php

```diff
@@ -77,32 +66,34 @@ public function isEmpty()
 
     public function tick()
     {
-        $time = $this->updateTime();
-        $timers = $this->timers;
-        $scheduler = $this->scheduler;
-
-        while (!$scheduler->isEmpty()) {
-            $timer = $scheduler->top();
+        // ensure timers are sorted so we can execute in order
+        if (!$this->sorted) {
+            $this->sorted = true;
+            asort($this->schedule);
+        }
 
-            if (!isset($timers[$timer])) {
-                $scheduler->extract();
-                $timers->detach($timer);
+        $time = $this->updateTime();
+        $schedule = $this->schedule;
```

Comment:
As `$this->schedule` is an array, by doing `$schedule = $this->schedule` you create a copy which looks like a waste of memory in order "to look good" at line 78.
If you use `foreach ($this->schedule as $id => $scheduled) {}` instead it should be fine (even if the list increases or decreases in number, since the iterator will operate with a local copy of it during the first loop).

Reply:
Good catch! This was a leftover from the original version and it's indeed safe to remove this :+1: Updated as per your suggestion, thank you.